tools/trap/ramdumpParser.py

```diff
@@ -866,20 +868,36 @@ def find_crash_point(log_file, elf):
     print('\nStack_address\t Symbol_address\t Symbol location  Symbol_name\t\tFile_name')
     os.system("python3 ../debug/debugsymbolviewer.py " + log_file + " " + str(g_app_idx) + " 0")
 
-    print('\nx. Miscellaneous information:')
+    print('\nh. Heap Region information:\n')
 
     with open(log_file) as searchfile:
-    # Parse the contents based on tokens in log file for heap corruption
         heap_corr = 0
+        for line in searchfile:
+            # Check if there is heap corruption or not
+            if 'Heap corruption detected' in line:
+                    heap_corr = 1
+        if (heap_corr == 1):
+            print(' -HEAP CORRUPTION DETECTED\n\n')
+        else:
+            print(' -NO HEAP CORRUPTION DETECTED\n\n')
+
+    with open(log_file) as searchfile:
+    # Parse the contents based on tokens in log file for heap corruption
         for line in searchfile:
             # Print the heap corruption data (if any)
+            if 'Checking kernel heap for corruption...' in line:
+                print("\033[F", line)
+            if 'Checking current app heap for corruption...' in line:
+                print("\033[F", line)
+            if 'No kernel heap corruption detected' in line:
```

Comment:
This condition cannot be happened concurrently with line 888.
So how about change to `else if` with line 888?